addons/sale_project/views/project_task_views.xml

```diff
@@ -231,8 +211,47 @@
                                 ('partner_id', '=', False),
                         ],
                     }"
-                />
+                >
+                    <field name="sale_line_id"
+                        groups="!sales_team.group_sale_salesman"
+                        string="Sales Order Item"
+                        options='{"no_open": True}'
+                        readonly="1"
+                        context="{'create': False, 'edit': False, 'delete': False}"
+                        attrs="{
+                            'invisible': [
+                                ('sale_line_id', '=', False),
+                            ],
+                        }"
+                    />
+                    <field name="sale_line_id"
+                        groups="sales_team.group_sale_salesman"
+                        string="Sales Order Item"
+                        options='{"no_create": True}'
+                        readonly="0"
+                        context="{
+                            'create': False, 'edit': False, 'delete': False,
+                            'with_price_unit': True,
+                            'form_view_ref': 'sale_project.sale_order_line_view_form_editable',
+                            'default_partner_id': partner_id,
+                            'default_company_id': company_id,
+                            'default_order_id': project_sale_order_id,
+                        }"
+                        placeholder="Non-billable"
+                        
+                    />
+                    <span 
+                        class="fa fa-exclamation text-warning oe_edit_only" 
+                        title="The sales order associated with this task has been canceled. We recommend either updating the sales order item or canceling this task in alignment with the cancellation of the sales order." 
+                        attrs="{
+                            'invisible': [
+                                ('sale_order_state', 'not in', ['cancel'])
```

Comment:
```suggestion
                                ('sale_order_state', '!=', 'cancel')
```

Reply:
Thanks @auon-odoo  for your suggestion,
changes are done